File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    // public int maxProfit(int[] prices) {
    //     int small = prices[0];
    //     int profit = 0;
    //     for (int i = 1;i<prices.length;i++){
    //         if (prices[i] > small){
    //             profit = Math.max(profit,prices[i] - small);
    //         }
    //         else{
    //             small = prices[i];
    //         }
    //     }

    //     return profit;
    // }

        public int maxProfit(int[] prices) {
            int smallest = prices[0];
            int ans = 0;
            for (int i = 1; i < prices.length;i++){
                if (smallest > prices[i]){
                    smallest = prices[i];
                    continue;
                }
                ans = Math.max(ans, prices[i] - smallest);
            }

            return ans;

    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curmin = prices[0]
        curmax = prices[0]
        profit = 0


        for price in prices:
            if curmin > price:
                curmin = price

            else:
                profit = max(profit, price - curmin)

        return profit